### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by tomyyyD**
Documentation in a README.md as there are only 4 functions to be documented at the moment.

### CODE REVIEW

**Comment by tomyyyD** [comment]
also removes the cursor input for rexDB as it is not needed any longer.

File: README.md [hunk 1]
```diff
@@ -1,3 +1,107 @@
...
+## table of contents
+- [Overview](#how-it-works)
+- [Methods](#methods)
+  - [\_\_init\_\_](#constructor-init)
+  - [log](#log)
+  - [get_data_at_time](#get_data_at_time)
+  - [get_data_at_range](#get_data_at_range)
+## How it works.
+
+rexDB works in a very straightforward manner. It works through the operating system file structure. The database is stored in a directory called db\_\<number\>, this is so that multiple databases could be stored in the same directory. inside the database folder is another set of folders and within those folders are the files that contain your entries. However, these files are unreadable as they are just structs packed into bytes.
```
Comment (thetazero) [inline README.md]: capital R

File: README.md [hunk 2]
```diff
@@ -1,3 +1,107 @@
...
+Each folder has a special file called a map, this map stores the start and end time of each file within the folder. This lets the query manager very easily ascertain if a given entry will be within a folder and if it is within a folder, which file it is in. This makes querying based on time much faster than querying based on other fields in the database.
+
+## Methods
```
Comment (thetazero) [inline README.md:17]: There should be a way to auto generate these docs, which is more convenient.
Although given that you have already done this, probably not worth it.

**Review by thetazero — CHANGES_REQUESTED** [review]

**Comment by thetazero** [comment]
The high level overview seems great

**Review by thetazero — APPROVED** [review]

**Pull request merged**